cranelift/filetests/filetests/isa/aarch64/simd-bitwise-compile.clif

```diff
@@ -257,27 +257,19 @@ block0:
 
 ; VCode:
 ; block0:
-;   ldr q5, pc+8 ; b 20 ; data.f128 0x0f0e0d0c0b0a09080706050403020100
-;   movz w1, #1
-;   and w3, w1, #7
-;   sub x5, xzr, x3
-;   dup v7.16b, w5
-;   ushl v0.16b, v5.16b, v7.16b
+;   ldr q1, pc+8 ; b 20 ; data.f128 0x0f0e0d0c0b0a09080706050403020100
+;   ushr v0.16b, v1.16b, #1
 ;   ret
 ; 
 ; Disassembled:
 ; block0: ; offset 0x0
-;   ldr q5, #8
+;   ldr q1, #8
```

Comment:
Orthogonal to your changes here but I'm surprised we haven't moved from inline constants we branch around to the `VCodeConstant` infrastructure; I guess I am confusing this with the x64 backend, where we do have that. Using the end-of-func constant pool here as well might be a nice improvement for a followup PR, especially if the data lands in the middle of an inner loop and pushes the body into another icache line or fetch cycle...

Reply:
Agreed! The fallback for `shuffle` does an inline load like this as well, and I'm curious enough to hopefully try to give this a stab in the near future